app/services/order_cycle_distributed_products.rb

```diff
@@ -42,6 +114,22 @@ def query_stock_with_overrides
                                       AND #{variant_not_on_demand} AND #{variant_in_stock} ) )"
   end
 
+  def query_with_tag_rules
+    "#{variant_not_overriden} OR ( #{variant_overriden}
+                                   AND ( #{override_not_hidden_by_rule}
+                                   OR #{override_shown_by_rule} ) )"
+  end
+
+  def override_not_hidden_by_rule
+    return "FALSE" unless overrides_to_hide.any?
+    "variant_overrides.id NOT IN (#{overrides_to_hide.join(',')})"
+  end
+
+  def override_shown_by_rule
+    return "FALSE" unless overrides_to_show.any?
+    "variant_overrides.id IN (#{overrides_to_show.join(',')})"
+  end
```

Comment:
if if the code s coupled with this class, we need to extract this code to a separate class. Maybe TagRuleApplicatorQuery? and inside both override_shown_by_rule and override_not_hidden_by_rule and all sub methods

Reply:
Already done ;)